board: arm: sparkfun_pro_micro_rp2040: Support JTAG

d3zd3z opened the pull request:
Add JTag debugger support for this board.  Although there is not a debug header, the SWD pins are accessible and the adaptor is not that difficult to wire up.  This will allow for debugging beyond just flashing, that the UF2 programming setup allows for.

d3zd3z committed:
boards: arm: sparkfun_pro_micro_rp2040: Support JTAG

Add SWD debugger support for this board.  Although there is not a debug
header, the SWD pins are accessible and the adaptor is not that difficult
to wire up.  This will allow for debugging beyond just flashing, that the
UF2 programming setup allows for.

Signed-off-by: David Brown <david.brown@linaro.org>

soburi commented on boards/arm/sparkfun_pro_micro_rp2040/doc/index.rst:
```diff
@@ -115,12 +115,14 @@ Flashing
-storage device. The UF2 file should be drag-and-dropped to the device, which
-will flash the Pro Micro RP2040.
+The Pro Micro board does make the SWD pins available on pads on the
+underside of the board. You can soldier to these pins, and use a JTag
```
Since SWD is not strictly JTAG, I think simply a debug adapter is better here.

d3zd3z replied:
I changed it to SWD debugger.

soburi commented on boards/arm/sparkfun_pro_micro_rp2040/board.cmake:
```diff
@@ -0,0 +1,39 @@
+
+# The adapter speed is expected to be set by interface configuration.
+# But if not so, set 2000 to adapter speed.
+board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
```
Also you need to copy
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/rpi_pico/support/openocd.cfg.

d3zd3z replied:
Done

nordicjm commented on boards/arm/sparkfun_pro_micro_rp2040/board.cmake:
```diff
@@ -0,0 +1,39 @@
+board_runner_args(uf2 "--board-id=RPI-RP2")
+board_runner_args(pyocd "--target=rp2040")
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```
think uf2 should come first in the list so it is used first, most people will be programming this using the uf2 bootloader

d3zd3z replied:
Isn't the default set above in the little if block? I can change that one to UF2. I'm not sure if inclusion order matters.

d3zd3z replied:
Ok, I'm pretty sure order doesn't matter. It always builds UF2, regardless of debug settings, as that it set by the defconfig.  And UF2 isn't a debug adaptor. I just copied this from all of the other rp2040 boards that are supported.

soburi replied:
> Ok, I'm pretty sure order doesn't matter. 

No it isn't.
We determine the default debug/flash setting with the order of runner, or following the setting of `board_set_runner_ifnset`/`board_set_runner_ifnset` if these are called.
And the runner setting know itself support flashing or debugging or both.
Since the debug terminals on this board are particularly difficult to use for usual, it seems easier to use settings that prioritize flash using uf2.

kartben commented on boards/sparkfun/pro_micro_rp2040/doc/index.rst:
```diff
@@ -115,12 +115,14 @@ Flashing
+By default, building an app for this board will generate a
+`build/zephyr/zephyr.uf2` file. If the Pro Micro RP2040 is powered on with
+the `BOOTSEL` button pressed, it will appear on the host as a mass storage
+device. The UF2 file should be drag-and-dropped to the device, which will
```
```suggestion
device. The UF2 file should be copied to the device, which will
```

kartben commented on boards/sparkfun/pro_micro_rp2040/doc/index.rst:
```diff
@@ -115,12 +115,14 @@ Flashing
-storage device. The UF2 file should be drag-and-dropped to the device, which
-will flash the Pro Micro RP2040.
+The Pro Micro board does make the SWD pins available on pads on the
+underside of the board. You can soldier to these pins, and use a JTag
```
```suggestion
underside of the board. You can solder to these pins, and use a JTag
```

kartben commented on boards/sparkfun/pro_micro_rp2040/support/openocd.cfg:
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2022 Tokita, Hiroshi <tokita.hiroshi@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+# Checking and set 'adapter speed'.
+# Set adapter speed that assigned by argument if not be seted.
```
Worth making a pass on the grammar/spelling (and apply the same changes wherever else this was, apparently, copied from)

d3zd3z committed:
boards: Spelling and grammar fix in openocd comment

This file has been copied a few times.  Clean up the grammar and spelling
in a comment in the file.

Signed-off-by: David Brown <david.brown@linaro.org>

d3zd3z commented:
@kartben, @soburi I think I've addressed your concerns. Any chance for a re-review?
